src/nodejs.js

```diff
@@ -41,31 +42,22 @@ const getResponse = resp => {
 }
 
 class StatusError extends Error {
-  constructor (res, ...params) {
+  constructor (res, body, ...params) {
     super(...params)
 
     Error.captureStackTrace(this, StatusError)
     this.message = `Incorrect statusCode: ${res.statusCode}`
     this.statusCode = res.statusCode
-    this.responseBody = new Promise((resolve) => {
-      const buffers = []
-      res.on('data', chunk => buffers.push(chunk))
-      res.on('end', () => resolve(Buffer.concat(buffers)))
-    })
+    this.responseBody = body || res
```

Comment:
So, instead of splitting the `onSuccess/FailureEncode` why not just create a separate API for body decoding the different formats and attach them as methods of `StatusError`.

Example:

```js
const get = bent(‘json’)

try {
  await get(‘https://site.com/data.json’)
} catch (e) {
  // json decode
  const json = await e.json()
  // string decode
  const str = await e.string()
  // binary decode
  const buf = await e.binary()
}
```

Also, keep in mind that any API for this we’ll need to replicate eventually for the browser. This API is much easier to replicate than the other one, these are almost identical to API’s already on fetch responses.

Reply:
I really like this API